### PR TITLE
Improve single file

### PR DESCRIPTION
**Pull request by TalMalka123**
- Add support to text and pdf files, in the single file view modal.
- Improve screen real estate usage within the single file view modal:
   - Increase modal size.
   - Decrease padding.
   - Make the top buttons section aligned with the X button, instead of being a separate line.
   - Make it possible to collapse and expand the metadata sidebar within the single file view modal.
   - Make files within the modal take up as much of the screen as possible.
- Left and right keyboard keys are now able to navigate files.
- Click outside the modal it will close it.
- The escape button will close the modal as well.
- The link to open the file in a new tab should be next to the filename, and it should have a tooltip.
- Add tooltips to the key and values in the metadataKeyValueList.
- Make buttons blurred automatically after clicking on it.
- Add download and annotate buttons with flags for each. (I added flags so we can deploy it without waiting for the annotate function, and also in case there will be issues with the download and we will want to be able to deploy without it).

### CODE REVIEW

File: src/components/dagshub/data-engine/PDFViewer/PDFViewer.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import 'react-pdf/dist/esm/Page/AnnotationLayer.css';
+import 'react-pdf/dist/esm/Page/TextLayer.css';
+import './PDFViewer.css';
+import type { PDFDocumentProxy } from 'pdfjs-dist';
+import React from 'react';
+
+pdfjs.GlobalWorkerOptions.workerSrc = new URL(
+  'pdfjs-dist/build/pdf.worker.min.js',
+  import.meta.url
+).toString();
```
Comment (simonlsk) [inline src/components/dagshub/data-engine/PDFViewer/PDFViewer.tsx:12]: Where is the `pdf.worker.min.js` located?

Comment (simonlsk) [reply src/components/dagshub/data-engine/PDFViewer/PDFViewer.tsx:12]: How does that work?

File: src/components/dagshub/data-engine/singleFileViewModal/ItemFallback.tsx [hunk 2]
```diff
@@ -6,12 +6,14 @@ export function ItemFallback({
   height,
   width,
   onClick,
-  disableHoverMode
+  disableHoverMode,
+    isError
```
Comment (simonlsk) [inline src/components/dagshub/data-engine/singleFileViewModal/ItemFallback.tsx:10]: Spaces

File: src/components/dagshub/data-engine/singleFileViewModal/ItemFallback.tsx [hunk 3]
```diff
@@ -33,7 +35,7 @@ export function ItemFallback({
         }
       }}
     >
-      <Icon icon={'outline-file'} width={24} height={24} fill={'#94A3B8'} />
+        {isError?<Icon icon={'file-x'} width={24} height={24} fill={'#94A3B8'} />:<Icon icon={'outline-file'} width={24} height={24} fill={'#94A3B8'} />}
```
Comment (simonlsk) [inline src/components/dagshub/data-engine/singleFileViewModal/ItemFallback.tsx:38]: Formatting.

File: src/components/dagshub/data-engine/singleFileViewModal/SingleFileViewFileRenderer.tsx [hunk 4]
```diff
@@ -1,51 +1,137 @@
...
+    const [textContent, setTextContent] = useState<string>('');
+    const [isError, setIsError] = useState<boolean>(false);
+
+    useEffect(() => {
+        setIsError(false);
+    }, [galleryFilePath, itemType])
+
+    useEffect(() => {
+        if (itemType === 'text') {
+            debugger;
```
Comment (simonlsk) [inline src/components/dagshub/data-engine/singleFileViewModal/SingleFileViewFileRenderer.tsx:25]: 👀

**Review by simonlsk — CHANGES_REQUESTED** [review]
LGTM, some comments.
I don't understand why the whole focus saga if you can just highlight buttons onClick, see if Anna is OK with that.

Bug:
When clicking next next ... in the modal story I get to a point I'm stuck between the csv and the png at the end.

File: public/assets/textFile.txt
[the remark this answers is not in the document]
Comment (TalMalka123) [reply public/assets/textFile.txt:1]: no, generated by chat gpt

**Comment by TalMalka123** [comment]
Button will be handled separately

File: src/components/dagshub/data-engine/PDFViewer/PDFViewer.tsx (hunk 1 above)
Comment (TalMalka123) [reply src/components/dagshub/data-engine/PDFViewer/PDFViewer.tsx:12]: i just followed the instructions in their readme file
https://github.com/wojtekmaj/react-pdf

**Review by simonlsk — APPROVED** [review]

**Pull request merged**